web/client/epics/timeline.js

```diff
@@ -546,6 +551,41 @@ export const rangeOnInitSelectLayer = (action$, { getState = () => { } } = {}) =
             Rx.Observable.of(initializeRange(!isNil(allowSnap) && !allowSnap ? "fullRange" : "now"))
         );
 
+/**
+ * Set time layers with settings
+ * on change of action with respect to the layers
+ * @param action$
+ * @param {function} getState returns the state
+ * @return {observable}
+ */
+export const setTimeLayersSetting = (action$, { getState = () => { } } = {}) =>
+    action$.ofType(INIT_TIMELINE, REMOVE_NODE, CHANGE_LAYER_PROPERTIES, UPDATE_NODE)
+        .switchMap(({config} = {}) => {
+            const state = getState();
+            let $observable = Rx.Observable.empty();
+            const timeLayers = timelineLayersSetting(state) || [];
+            if (!isEmpty(config) && !isEmpty(timeLayers)) {
+                // Skip time layer update on initialization
+                // as layer data is updated from map config
+                return $observable;
+            }
+            const layers = layersWithTimeDataSelector(state)?.filter(l => l.visibility);
+            if (!isEmpty(layers)) {
+                let layersSetting = layers.map(layer => {
+                    const id = layer.id;
+                    const timeLayer = timeLayers.find(l => l.id === id);
+                    return {title: layer.title, id, checked: isEmpty(timeLayer) ? true : timeLayer.checked};
```

Comment:
You should not store the title of the layer here. Only the `id` and the properties you want to add should be there.
(also a map `{ ["LAYER_ID"]: {props...}}` may make sense.
The property should be "hideInTimeline" or similar, because "checked" is a UX tool condition, not representing the semantics. Moreover I suggest "hide" instead of "show" because "show" (checked) is the default. 
Using hide you don't need to initialize anything. And you may be able to store only the information when the user changes something.. 

-In layer settings, set as unchecked only the items with "hideInTimeline" = true.
- when the user unchecks the item, you can add / update the item in the settings. hideInTimeline: true
- when the user checks again the item, you can update it with hideInTimeline: "false".